### PR TITLE
Refactor chart data.

### PR DESCRIPTION
**Pull request by polina-c**
1. Remove live vs offline data from chart memory timeline.
2. Factor out chart data to separate class.

No functional changes.

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_pane_controller.dart [hunk 1]
```diff
@@ -34,13 +34,22 @@ class MemoryChartPaneController extends DisposableController
     );
   }
 
+  Map<String, dynamic> toJson() {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_pane_controller.dart:37]: nit: class member order: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#order-other-class-members-in-a-way-that-makes-sense

File: packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_data.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  final _displayInterval =
+      ValueNotifier<ChartInterval>(ChartInterval.theDefault);
+  set displayInterval(ChartInterval interval) {
+    _displayInterval.value = interval;
+  }
+
+  ValueListenable<bool> get isLegendVisible => _legendVisibleNotifier;
+  final _legendVisibleNotifier = ValueNotifier<bool>(true);
+  bool toggleLegendVisibility() =>
+      _legendVisibleNotifier.value = !_legendVisibleNotifier.value;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_data.dart:35]: nit: I don't know if it is important to save this data in the offline export

Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_data.dart:35]: I save everything but state configured in preferences.
It is cheap and create good UX. User will like to see the same thing, without having to re-hide legend.

**Review by kenzieschmoll — APPROVED** [review]
a couple nits

File: packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_pane_controller.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_pane_controller.dart:37]: They say 'If no particular order is obvious'.

fromJson and toJson are better readable when close to each other.

**Pull request merged**